students/arkadiusz_kasprzyk/lesson_05_lists/comma_code.py

```diff
@@ -0,0 +1,30 @@
+def comma_code( lst ):
+    """
+    Parameters
+    ----------
+    lst: list
+
+    Returns
+    -------
+    string constructed from elements of the `lst` by joining them with ', '
+    and the last two with ' and '.
+
+    Examples
+    --------
+    print(comma_code(['apples', 'bananas', 'tofu', 'cats', 'dogs']))
+    print(comma_code(['apples', 'bananas', 'tofu']))
+    print(comma_code(['apples', 'bananas']))
+    print(comma_code(['apples']))
+    print(comma_code([]))
+    """
+
+    s = ''
+    l = len(lst)
+
+    if l > 0:
+        s += str(lst[0])
+        if l > 1:
+            sep = [', '] * (l - 2) + [' and ']
+            for k in range(l - 1):
+                s += (sep[k] + str(lst[k + 1]))
+    return s
```

Comment:
I find it very confusing, especially since the naming is too short. Why not:
```
if len(lst) > 1:
  *all_but_last, last = lst
  return ', '.join(all_but_last) + ', and ' + last
else:
  return lst
```

Reply:
OK, but the second line `*all_but_last, last = lst` is tricky if you don't know it's possible to use `*` this way. I can't remember any such example before. Again, thanks for the tip!

I suffer a lot inventing names and unfortunatelly, I'm matematician 😁 so I love one-letter names. Sorry... I understand the problem in larger projects but in few-lines code... Never mind! I'll try to invent better names.

Reply:
That's a new thing with Python3, I think it's the clearest form of head and tail from functional languages. I never used it in real code yet and it's not obvious.